core/server/api/v2/utils/serializers/output/utils/mapper.js

```diff
@@ -0,0 +1,52 @@
+const utils = require('../../../');
+const url = require('./url');
+const date = require('./date');
+
+const mapPost = (model, frame) => {
+    const jsonModel = model.toJSON(frame.options);
+
+    url.forPost(model.id, jsonModel, frame.options);
+
+    if (utils.isContentAPI(frame)) {
+        date.forPost(jsonModel);
+    }
+
+    if (frame.options && frame.options.withRelated) {
+        frame.options.withRelated.forEach((relation) => {
+            // @NOTE: this block also decorates primary_tag/primary_author objects as they
+            // are being passed by reference in tags/authors. Might be refactored into more explicit call
+            // in the future, but is good enough for current use-case
+            if (relation === 'tags' && jsonModel.tags) {
+                jsonModel.tags = jsonModel.tags.map(tag => url.forTag(tag.id, tag));
+
+                if (utils.isContentAPI(frame)) {
+                    jsonModel.tags = jsonModel.tags.map(tag => date.forTag(tag));
```

Comment:
I am getting `tags=[null,null]` while testing.
This is probably because `forTag` does not return anything.